src/NavDropdown.js

```diff
@@ -1,47 +1,32 @@
 import React from 'react';
-import classNames from 'classnames';
-import uncontrollable from 'uncontrollable';
-import DropdownBase from './DropdownBase';
-import DropdownToggle from './DropdownToggle';
-import DropdownMenu from './DropdownMenu';
+import Dropdown from './Dropdown';
 
-class NavDropdown extends DropdownBase {
-  constructor(props) {
-    super(props);
-
-    this.useAnchor = true;
-  }
+class NavDropdown extends React.Component {
 
   render() {
-    let {
-      toggle,
-      menu,
-      open
-    } = this.extractChildren();
-
-    const rootClasses = {
-      open,
-      dropdown: !this.props.dropup,
-      dropup: this.props.dropup
-    };
+    let { children, title, noCaret, bsStyle, ...props } = this.props;
 
     return (
-      <li className={classNames(this.props.className, rootClasses)}>
-        {toggle}
-        {menu}
-      </li>
+      <Dropdown {...props} componentClass='li'>
+        <Dropdown.Toggle
+          useAnchor
+          bsStyle={bsStyle}
+          disabled={props.disabled}
+          noCaret={noCaret}
+        >
+          {title}
+        </Dropdown.Toggle>
+        <Dropdown.Menu>
+          {children}
+        </Dropdown.Menu>
```

Comment:
Ok, delayed understanding. You are proposing that these methods are not customizable, but if you want to roll your own with the Dropdown component you can. I'm catching your drift now.